### PR TITLE
Patch metadata

### PR DESCRIPTION
**Pull request by DanielElisenberg**
# PATCH METADATA

Functionality for the patch metadata case.

### What's changed?
* Implemented metadata_builder_service.py for this case
* Updated app.py to send jobs to the correct builder
* Made appropriate unit tests for the new service
* Updated the transformer to allow a lack of temporality values

### What's next?
* To avoid crowding this PR with too much refactoring I kept it as simple as possilbe. Will post a new PR with some refactoring of code duplication. There are 2 services now, and they share a lot of common code that could be refactored to steps instead.

### CODE REVIEW

File: dataset_builder/domain/metadata_builder_service.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        job_service_adapter.update_status(job_id, 'initiated')
+        config = config_provider.get_config()
+        working_dir = config['WORKING_DIR']
+        output_dir = config['OUTPUT_DIR']
+        input_dir = config['INPUT_DIR']
+
+        log_config.setup_logger_for_import_pipeline()
+        logger.info(f'Starting to build metadata {dataset_name}')
+
+        with open(
```
Comment (pawbu) [inline dataset_builder/domain/metadata_builder_service.py:36]: Why load & dump and not copy with shutil?

File: dataset_builder/steps/dataset_transformer.py [hunk 2]
```diff
@@ -35,8 +35,12 @@ def _represented_variables_from_description(description: str,
         {
             "description": description,
             "validPeriod": {
-                "start": _days_since_epoch(start),
-                "stop": _days_since_epoch(stop)
+                "start": (
```
Comment (pawbu) [inline dataset_builder/steps/dataset_transformer.py:38]: 💯

File: dataset_builder/steps/dataset_transformer.py [hunk 3]
```diff
@@ -251,8 +254,12 @@ def transform_metadata(metadata_file_path: str) -> str:
         metadata = json.load(json_file)
     logger.info(f'Transforming metadata {metadata_file_path}')
 
-    start = metadata['dataRevision']['temporalCoverageStart']
-    stop = metadata['dataRevision']['temporalCoverageLatest']
+    # These values are found by going through the data file.
```
Comment (pawbu) [inline dataset_builder/steps/dataset_transformer.py:257]: 💯

**Review by pawbu — APPROVED** [review]
Looks good, one question

File: dataset_builder/domain/metadata_builder_service.py (hunk 1 above)
Comment (DanielElisenberg) [reply dataset_builder/domain/metadata_builder_service.py:36]: I'll fix 👍🏻 I originally did something with the json during the move. After removing the processing I didn't rethink the copying 😅

**Pull request merged**